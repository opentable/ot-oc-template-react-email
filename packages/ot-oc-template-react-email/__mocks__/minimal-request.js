const state = {};
module.exports = (opts, cb) => {
  var jsText =
    state.jsText ||
    `var oc = {reactComponents: {}}; oc.reactComponents['666'] = class Hello extends React.Component {
    render() {
      return React.createElement("div", null, "Hello World");
    }
  }`;
  return cb(null, jsText);
};

module.exports.__setResponse = jsText => {
  state.jsText = jsText;
};
