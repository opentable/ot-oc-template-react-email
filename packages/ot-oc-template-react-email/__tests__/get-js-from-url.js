const getJSFromUrl = require("../lib/to-be-published/get-js-from-url");
const React = require("react");

describe("When the module is called", () => {
  const predicate = getJSFromUrl({
    globals: { React },
    url: "cdn/path/to/file",
    key: "666",
    extractor: jest.fn((key, context) => context.oc.reactComponents[key])
  });

  test("Should correctly initialize a function", () => {
    expect(predicate).toBeInstanceOf(Function);
    expect(predicate.length).toBe(1);
  });

  describe("and the returned function is invoked", () => {
    test("should correctly return the evaluated js", done => {
      predicate((err, res) => {
        expect(err).toBe(null);
        expect(res).toBeInstanceOf(Function);
        expect(res.toString()).toMatchSnapshot();
        done();
      });
    });

    describe("but the evaluated js causes an error", () => {
      const jsError = "throw new Error('Error in template');";
      beforeEach(() => {
        require("minimal-request").__setResponse(jsError);
      });

      test("should send the error to the callback", done => {
        predicate((err, res) => {
          expect(err.message).toBe("Error in template");
          done();
        });
      });
    });
  });
});
