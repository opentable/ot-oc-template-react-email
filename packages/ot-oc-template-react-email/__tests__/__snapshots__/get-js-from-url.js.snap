// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the module is called and the returned function is invoked should correctly return the evaluated js 1`] = `
"class Hello extends React.Component {
    render() {
      return React.createElement(\\"div\\", null, \\"Hello World\\");
    }
  }"
`;
